drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -206,15 +227,47 @@ static int siwx917_scan(const struct device *dev, struct wifi_scan_params *z_sca
 		return -EBUSY;
 	}
 
-	/* The enum values are same, no conversion needed */
-	sl_scan_config.type = z_scan_config->scan_type;
+	if (z_scan_config->scan_type == WIFI_SCAN_TYPE_ACTIVE) {
+		sl_scan_config.type = SL_WIFI_SCAN_TYPE_ACTIVE;
+		if (z_scan_config->dwell_time_active) {
+			ret = sl_si91x_configure_timeout(SL_SI91X_CHANNEL_ACTIVE_SCAN_TIMEOUT,
```

Comment:
This function looks to have a side effect. So, if you call `siwx917_scan()` with `dwell_time_active != 0`, then `dwell_time_active == 0`, previous `dwell_time_active` will continue to apply.